sbat.c

```diff
@@ -335,4 +335,62 @@ parse_sbat_var(list_t *entries)
 	FreePool(data);
 	return EFI_INVALID_PARAMETER;
 }
+
+EFI_STATUS
+init_sbat()
+{
+	EFI_STATUS efi_status = EFI_SUCCESS;
+	UINT32 attributes = 0;
+
+	CHAR8 *sbat;
+	UINTN sbatsize = 4096;
+
+	sbat = AllocateZeroPool(sbatsize);
+
+	/* read and get check attributes */
+	efi_status = gRT->GetVariable(L"SBAT", &SHIM_LOCK_GUID, &attributes,
+			&sbatsize, sbat);
+
+	/*
+	 * If attributes and the initial part of the version match, then
+	 * there is no need to re-write it. If it is not set or the read
+	 * failed, then attributes won't match.
+	 */
+	if (((attributes != (EFI_VARIABLE_NON_VOLATILE |
+		EFI_VARIABLE_BOOTSERVICE_ACCESS)) ||
+		(attributes != (EFI_VARIABLE_NON_VOLATILE |
+				EFI_VARIABLE_BOOTSERVICE_ACCESS |
+				EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS)))
+			|| (strncmp(sbat, SBAT_INIT, sizeof(SBAT_SIG) - 1) != 0)) {
```

Comment:
Whey is `SBAT_INIT` used here instead of just `SBAT_SIG` ?

Reply:
Don't we also want to compare the version here?

Reply:
After staring at variously ugly ways to do the date stamp and version comparison, I realized that strcmp() will actually do the right thing.

and yes, I should use SBAT_SIG or rather SBAT_VAR_SIG